components/playlist/browser/resources/components/header.tsx

```diff
@@ -120,29 +145,90 @@ export default function Header ({ playlistId }: HeaderProps) {
   const isDefaultPlaylist = playlist?.id === 'default'
   if (contextualMenuItems && !isDefaultPlaylist) {
     contextualMenuItems.unshift({
-      name: 'Rename',
+      name: getLocalizedString('bravePlaylistContextMenuRenamePlaylist'),
       iconName: 'edit-box',
-      onClick: () => {}
+      onClick: () => {
+        getAllActions().setPlaylistEditMode(PlaylistEditMode.RENAME)
+      }
     })
     contextualMenuItems.push({
-      name: 'Delete playlist',
+      name: getLocalizedString('bravePlaylistContextMenuDeletePlaylist'),
       iconName: 'trash',
       onClick: () => {
-        getPlaylistAPI().showRemovePlaylistUI(playlistId!)
+        getPlaylistAPI().showRemovePlaylistUI(playlistId)
       }
     })
   }
 
   const totalDuration = useTotalDuration(playlist)
   const totalSize = useTotalSize(playlist)
+  const playlistEditMode = usePlaylistEditMode()
 
-  return (
-    <HeaderContainer>
-      {playlist ? (
+  const [hasNewName, setHasNewName] = React.useState(true)
+  const inputElementRef = React.useRef<HTMLInputElement>(null)
```

Comment:
Optional: I think you can get rid of the `useRef` here, if you store the `newName` text:

```tsx
const [newName, setNewName] = React.useState(playlist.name)
const hasNewName = !!newName

<input value={newName} ... onChange={e => setNewName(e.target.value)}/>
```

Reply:
That's much better!